internal/command/deploy/deploy.go

```diff
@@ -193,22 +193,22 @@ func New() (cmd *cobra.Command) {
 
 func run(ctx context.Context) error {
 	io := iostreams.FromContext(ctx)
+	appName := appconfig.NameFromContext(ctx)
 
 	hook := ctrlc.Hook(func() {
 		metrics.FlushMetrics(ctx)
 	})
 
 	defer hook.Done()
 
-	tp, err := tracing.InitTraceProvider(ctx)
+	tp, err := tracing.InitTraceProvider(ctx, appName)
```

Comment:
Could you make this optional? We don't obtain the app name for the launch command until a couple of lines late into the command, so it doesn't work either.